packages/at_client/lib/src/client/at_client_impl.dart

```diff
@@ -881,6 +881,113 @@ class AtClientImpl implements AtClient, AtSignChangeListener {
     }
   }
 
+  @override
+  Future<void> isSecondaryReset() async {
+    _logger.finer('Performing Remote Secondary reset check');
+    var localPublicKey =
+        await getLocalSecondary()?.getEncryptionPublicKey(getCurrentAtSign()!);
+
+    LookupVerbBuilder lookupBuilder = LookupVerbBuilder()
+      ..atKey = 'publickey' //AT_ENCRYPTION_PUBLIC_KEY
+      ..sharedBy = getCurrentAtSign();
+    String? remotePublicKey =
+        await getRemoteSecondary()?.executeVerb(lookupBuilder);
+    // secondary response is in format 'data:publickey'. Removing 'data:' from response
+    remotePublicKey = remotePublicKey?.replaceFirst('data:', '');
+
+    if (localPublicKey.isNull) {
+      _logger.info('Could not fetch EncryptionPublicKey from LocalSecondary.'
+          ' Unable to complete reset check');
+      return;
+    }
+
+    // TODO should there be a check for remote enc_key being null ?
+
+    if (localPublicKey != remotePublicKey) {
+      _logger.info('EncryptionPublicKey on LocalSecondary: $localPublicKey');
+      _logger.info('EncryptionPublicKey on RemoteSecondary: $remotePublicKey');
+      _logger.shout(
+          'AtEncryptionPublicKey on local secondary and remote secondary are different.'
+          '\nThis indicates remote secondary has been reset.'
+          '\nPlease delete localStorage and restart the client');
+      _logger.info('To delete localSecondary,'
+          ' call AtClientImpl.deleteLocalSecondary() with user consent');
+      throw AtResetException('Remote secondary has been reset');
+    }
+    _logger.finer('Secondary is not reset. Status ok');
+  }
+
+  void deleteLocalSecondaryStorageWithConsent(
+      String commitLogDirectoryPath, String hiveStorageDirectoryPath,
+      {required bool userConsentToDeleteLocalSecondaryStorage}) {
+    _logger.shout(
+        'Consent to delete LocalSecondary storage received: $userConsentToDeleteLocalSecondaryStorage');
+    if (userConsentToDeleteLocalSecondaryStorage) {
+      _logger
+          .info('Deleting CommitLog local storage at: $commitLogDirectoryPath');
+      _deleteLocalCommitLog(commitLogDirectoryPath);
+      _logger.info('Deleting hive local storage at: $hiveStorageDirectoryPath');
+      _deleteLocalHiveStorage(hiveStorageDirectoryPath);
+    }
+  }
+
+  void _deleteLocalCommitLog(String commitLogDirectoryPath) {
```

Comment:
have a single method instead of two methods for deleting commit log and main storage. 
e.g _deleteStorage(String storageDirectory, String storageType)//storageType can be commitlog or main

Reply:
change made as part of [6b0efb8](https://github.com/atsign-foundation/at_client_sdk/pull/1033/commits/6b0efb847d7cd4d3e5e080a026622bb83b9402ad)